transformer/dockerfilegenerator/nodejsdockerfiletransformer.go

```diff
@@ -249,9 +267,11 @@ func (t *NodejsDockerfileGenerator) Transform(newArtifacts []transformertypes.Ar
 			props = t.Spec.NodeVersions[idx]
 		}
 		nodejsConfig := NodejsTemplateConfig{
-			Build:                 build,
-			Port:                  port,
-			NodeVersion:           nodeVersion,
+			Build:       build,
+			Port:        port,
+			NodeVersion: nodeVersion,
+			// NodeTag:               nodeTag,
+			NodeMajorVersion:      strings.TrimPrefix(semver.Major(nodeVersion), "v"),
```

Comment:
We might not have all the node versions in the images, isn't it? So we might want to have a list of versions that are present and go to the nearest that fits the version criteria.

Reply:
Yes, currently there are 4 node versions in the ubi8/node images (10, 12, 14 and 16) that can be pulled. 

```console
docker run --rm -it -p 8080:8080 registry.access.redhat.com/ubi8/nodejs-10 bash
bash-4.4$ node -v
v10.24.0
```

```console
docker run --rm -it -p 8080:8080 registry.access.redhat.com/ubi8/nodejs-12 bash
bash-4.4$ node -v
v12.22.5
```

```console
docker run --rm -it -p 8080:8080 registry.access.redhat.com/ubi8/nodejs-14 bash
bash-4.4$ node -v
v14.20.1
```

```console
docker run --rm -it -p 8080:8080 registry.access.redhat.com/ubi8/nodejs-16 bash
bash-4.4$ node -v
v16.17.1
```

We have the updated list of available versions here- https://github.com/Akash-Nayak/move2kube/blob/13141078de0de81e2c705d85b31f162d06772933/assets/built-in/transformers/dockerfilegenerator/mappings/nodeversions.yaml#L6 and we select the node version that fits the version criteria here https://github.com/Akash-Nayak/move2kube/blob/03df5e77bd6734eb18dfd6645b3c108549655203/transformer/dockerfilegenerator/nodejsdockerfiletransformer.go#L222.